cg/meta/workflow/raredisease.py

```diff
@@ -149,11 +152,58 @@ def get_managed_variants(self) -> list[str]:
         return self._get_managed_variants(genome_build=GenePanelGenomeBuild.hg19)
 
     def get_workflow_metrics(self, sample_id: str) -> dict:
+        """Return Raredisease workflow metric conditions for a sample."""
         sample: Sample = self.status_db.get_sample_by_internal_id(internal_id=sample_id)
-        metric_conditions: dict[str, dict[str, Any]] = dict(RAREDISEASE_METRIC_CONDITIONS)
-        self.set_order_sex_for_sample(sample, metric_conditions)
+        if "-" not in sample_id:
+            metric_conditions: dict[str, dict[str, Any]] = RAREDISEASE_METRIC_CONDITIONS.copy()
+            self.set_order_sex_for_sample(sample, metric_conditions)
+        else:
+            metric_conditions = RAREDISEASE_PARENT_PEDDY_METRIC_CONDITION.copy()
         return metric_conditions
 
+    def _get_sample_pair_patterns(self, case_id: str) -> list[str]:
+        """Return sample-pair patterns for searching in MultiQC."""
+        sample_ids: list[str] = list(self.status_db.get_sample_ids_by_case_id(case_id=case_id))
+        pairwise_patterns: list[str] = [
+            f"{sample1}-{sample2}" for sample1, sample2 in permutations(sample_ids, 2)
+        ]
+        return pairwise_patterns
+
+    def get_parent_error_ped_check_metric(
+        self, pair_sample_ids: str, multiqc_raw_data: dict[dict]
+    ) -> MetricsBase | None:
+        """Return the parsed metrics for pedigree error given a concatenated pair of sample ids."""
+        metric_name: str = "parent_error_ped_check"
+        peddy_metrics: dict[str, dict] = multiqc_raw_data["multiqc_peddy"]
+        if sample_pair_metrics := peddy_metrics.get(pair_sample_ids, None):
+            return self.get_multiqc_metric(
+                metric_name=metric_name,
+                metric_value=sample_pair_metrics[metric_name],
+                metric_id=pair_sample_ids,
+            )
+
+    def get_multiqc_json_metrics(self, case_id: str) -> list[MetricsBase]:
+        """Return a list of the metrics specified in a MultiQC json file."""
+        multiqc_json: MultiqcDataJson = self.get_multiqc_data_json(case_id=case_id)
+        metrics = []
+        for search_pattern, metric_id in self.get_multiqc_search_patterns(case_id).items():
+            metrics_for_pattern: list[MetricsBase] = (
+                self.get_metrics_from_multiqc_json_with_pattern(
+                    search_pattern=search_pattern,
+                    multiqc_json=multiqc_json,
+                    metric_id=metric_id,
+                    exact_match=self.is_multiqc_pattern_search_exact,
+                )
+            )
+            metrics.extend(metrics_for_pattern)
+        for sample_pair in self._get_sample_pair_patterns(case_id):
+            if parent_error_metric := self.get_parent_error_ped_check_metric(
+                pair_sample_ids=sample_pair, multiqc_raw_data=multiqc_json.report_saved_raw_data
```

Comment:
Is there a way of baking this into the `self.get_metrics_from_multiqc_json_with_pattern`? Seems to me as if we are getting metrics from the multiqc json using a pattern.